src/test/regress/sql/remove_temp_schemas.sql

```diff
@@ -0,0 +1,49 @@
+CREATE OR REPLACE FUNCTION gp_execute_on_server(content int, query text)
+returns text language C as '$libdir/regress.so', 'gp_execute_on_server';
+
+-- First, handle master temp schemas
+DO $$
+DECLARE
+	nsp pg_namespace.nspname%type;
+BEGIN
+	FOR nsp IN
+		SELECT nspname
+		FROM (
+			SELECT nspname
+			FROM gp_dist_random('pg_namespace')
+			UNION
+			SELECT nspname
+			FROM pg_namespace
+		) n
+		LEFT JOIN pg_stat_activity ON
+			sess_id = regexp_replace(nspname, 'pg(_toast)?_temp_', '')::int
+		WHERE sess_id is null AND nspname ~ '^pg(_toast)?_temp_[1-9]{1}[0-9]*'
+	LOOP
+		EXECUTE format('DROP SCHEMA IF EXISTS %I CASCADE', nsp);
+	END LOOP;
+END $$;
+
+-- Then, handle segment temp schemas
+DO $$
+DECLARE
+	schema_rec record;
+BEGIN
+	FOR schema_rec IN
+		SELECT nspname, gp_segment_id
+		FROM (
+			SELECT nspname, gp_segment_id
+			FROM gp_dist_random('pg_namespace')
+			UNION
+			SELECT nspname, gp_segment_id
+			FROM pg_namespace
+		) n
+		LEFT JOIN pg_stat_activity ON
+			sess_id = regexp_replace(nspname, 'pg(_toast)?_temp_', '')::int
+		WHERE sess_id is null AND nspname ~ '^pg(_toast)?_temp_[1-9]{1}[0-9]*'
+	LOOP
+		PERFORM gp_execute_on_server(schema_rec.gp_segment_id,
+			format('DROP SCHEMA IF EXISTS %I CASCADE', schema_rec.nspname));
```

Comment:
```suggestion
		if schema_rec.gp_segment_id = -1 then
			EXECUTE format('DROP SCHEMA IF EXISTS %I CASCADE', schema_rec.nspname);
		else
			PERFORM gp_execute_on_server(schema_rec.gp_segment_id,
				format('DROP SCHEMA IF EXISTS %I CASCADE', schema_rec.nspname));
		end if;
```

Reply:
hm... gp_execute_on_server with -1 as first agruments executes sql on coordinator, then maybe remove "First, handle master temp schemas"?

Reply:
I got the the error when calling it with -1
ERROR:  AssignTransactionId() called by Segment Entry DB Singleton process (xact.c:670)  (entry db 127.0.0.1:7000 pid=1210835) (xact.c:670)